x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -492,11 +492,12 @@ export class Authenticator {
   private calculateExpiry(
     existingSession: ProviderSession | null
   ): { idleTimeoutExpiration: number | null; lifespanExpiration: number | null } {
-    let lifespanExpiration = this.lifespan && Date.now() + this.lifespan;
+    const now = Date.now();
+    let lifespanExpiration = this.lifespan && now + this.lifespan.asMilliseconds();
```

Comment:
__nit:__
```suggestion
const lifespanExpiration =
  existingSession?.lifespanExpiration && this.lifespan
    ? existingSession.lifespanExpiration
    : this.lifespan && now + this.lifespan.asMilliseconds();
```

Or maybe even like that (not sure if we need `&& this.lifespan` when we have `existingSession.lifespanExpiration`, would be great if you can add a clarifying comment here):
```suggestion
const lifespanExpiration = existingSession?.lifespanExpiration
  ? existingSession.lifespanExpiration
  : this.lifespan && now + this.lifespan.asMilliseconds();
```

Reply:
The reason it's written this way is because we need to account for two scenarios:
1. If a server config used `lifespan` and then removes it, all old/existing sessions should be updated to remove a `lifespanExpiration`
2. If a server config did _not_ use `lifespan` and then adds it, all old/existing sessions should be updated to add a `lifespanExpiration`

If you remove `&& this.lifespan` then scenario 1 would fail. These are described in tests in `authenticator.test.ts` but I'll add a comment too.